misc/loop/cmd/cmd_serve.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/docker/docker/client"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/sirupsen/logrus"
+)
+
+type serveCfg struct {
+	rpcAddr        string
+	traefikGnoFile string
+	backupDir      string
+	hostPWD        string
+}
+
+type serveService struct {
+	cfg serveCfg
+
+	// TODO(albttx): put getter on it with RMutex
+	portalLoop *snapshotter
+
+	portalLoopURL string
+}
+
+func (c *serveCfg) RegisterFlags(fs *flag.FlagSet) {
+	if os.Getenv("HOST_PWD") == "" {
+		os.Setenv("HOST_PWD", os.Getenv("PWD"))
+	}
+
+	if os.Getenv("BACKUP_DIR") == "" {
+		os.Setenv("BACKUP_DIR", "./backups")
+	}
+
+	if os.Getenv("RPC_URL") == "" {
+		os.Setenv("RPC_URL", "http://rpc.portal.gno.local:26657")
+	}
+
+	if os.Getenv("PROM_ADDR") == "" {
+		os.Setenv("PROM_ADDR", ":9090")
+	}
+
+	if os.Getenv("TRAEFIK_GNO_FILE") == "" {
+		os.Setenv("TRAEFIK_GNO_FILE", "./traefik/gno.yml")
+	}
+
+	fs.StringVar(&c.rpcAddr, "rpc", os.Getenv("RPC_URL"), "tendermint rpc url")
+	fs.StringVar(&c.traefikGnoFile, "traefik-gno-file", os.Getenv("TRAEFIK_GNO_FILE"), "traefik gno file")
+	fs.StringVar(&c.backupDir, "backup-dir", os.Getenv("BACKUP_DIR"), "backup directory")
+	fs.StringVar(&c.hostPWD, "pwd", os.Getenv("HOST_PWD"), "host pwd (for docker usage)")
```

Comment:
Can we use this instead? https://pkg.go.dev/github.com/peterbourgon/ff/v3#WithEnvVars